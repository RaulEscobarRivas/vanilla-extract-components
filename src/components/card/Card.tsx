import React from "react";
import classNames from "classnames";
import { useAsProp } from "../../hooks/useAsProp";
import { cardStyle, radiusStyle, hoverStyle } from "./Card.css";
import { CardProps } from "./types";

export function Card({ radius = "sm", withHover = false, ...rest }: CardProps) {
  const [Tag, props] = useAsProp(
    { as: "div", ...rest },
    {
      className: classNames(cardStyle, radiusStyle[radius], {
        [hoverStyle]: withHover,
      }),
    }
  );

  return <Tag {...props} />;
}
