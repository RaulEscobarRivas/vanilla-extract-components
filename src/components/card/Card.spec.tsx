import React from "react";
import { render } from "@testing-library/react";
import { Card } from "./Card";

describe("Card", () => {
  it("should render default style", () => {
    const { container } = render(<Card />);

    expect(container).toMatchSnapshot();
  });

  it("should render custom class name", () => {
    const { container } = render(<Card className="raulito" />);

    expect(container).toMatchSnapshot();
  });

  it("should render custom span tag", () => {
    const { container } = render(<Card as="span" />);

    expect(container).toMatchSnapshot();
  });

  it("should render md radius style", () => {
    const { container } = render(<Card radius="md" />);

    expect(container).toMatchSnapshot();
  });

  it("should render lg radius style", () => {
    const { container } = render(<Card radius="lg" />);

    expect(container).toMatchSnapshot();
  });

  it("should render xl radius style", () => {
    const { container } = render(<Card radius="xl" />);

    expect(container).toMatchSnapshot();
  });

  it("should render withHover style", () => {
    const { container } = render(<Card withHover />);

    expect(container).toMatchSnapshot();
  });

  it("should render variant styles", () => {
    const { container } = render(<Card as="section" radius="xl" withHover />);

    expect(container).toMatchSnapshot();
  });
});
