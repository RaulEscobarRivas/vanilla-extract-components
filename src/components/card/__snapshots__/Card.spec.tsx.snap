// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render custom class name 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_sm__1s6a79j4 raulito"
  />
</div>
`;

exports[`Card should render custom span tag 1`] = `
<div>
  <span
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_sm__1s6a79j4"
  />
</div>
`;

exports[`Card should render default style 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_sm__1s6a79j4"
  />
</div>
`;

exports[`Card should render lg radius style 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_lg__1s6a79j2"
  />
</div>
`;

exports[`Card should render md radius style 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_md__1s6a79j3"
  />
</div>
`;

exports[`Card should render variant styles 1`] = `
<div>
  <section
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_xl__1s6a79j1 Card_hoverStyle__1s6a79j5"
  />
</div>
`;

exports[`Card should render withHover style 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_sm__1s6a79j4 Card_hoverStyle__1s6a79j5"
  />
</div>
`;

exports[`Card should render xl radius style 1`] = `
<div>
  <div
    class="Card_cardStyle__1s6a79j0 Card_radiusStyle_xl__1s6a79j1"
  />
</div>
`;
